src/python/WMCore/MicroService/MSOutput/RelValPolicy.py

```diff
@@ -102,22 +119,95 @@ def _validatePolicy(self, policyDesc, validDBSTiers, validDiskRSEs):
             msg = "A 'default' key must be defined with default destinations."
             raise RelValPolicyException(msg) from None
 
-    def _convertPolicy(self, policyDesc):
+    def _validateLifetimePolicy(self, lifeTDesc):
         """
-        Maps the RelVal data policy to a flat dictionary key'ed by datatiers
-        :param policyDesc: list of dictionaries with the policy definition
-        :return: a dictionary with a map of the RelVal policy
+        This method validates the lifetime RelVal policy data structure.
+
+        [{"releaseType": "pre", "lifetimeSecs": 120},
+         {"releaseType": "default", "lifetimeSecs": 360}]
+        :param lifeTDesc: list of dictionaries with the lifetime policy definition
+        :return: nothing, but it will raise an exception if any validation fails
+        """
+        if not isinstance(lifeTDesc, list):
+            msg = "The RelVal lifetime output data placement policy is not in the expected data type. "
+            msg += "Type expected: list, while the current data type is: {}. ".format(type(lifeTDesc))
+            msg += "This critical ERROR must be fixed."
+            raise RelValPolicyException(msg) from None
+
+        # policy must have a default/fallback destination for non pre-releases
+        setRelTypes = set()
+        expRelTypesKeys = {"pre", "default"}
+        for item in lifeTDesc:
+            if not isinstance(item.get('releaseType', None), str):
```

Comment:
I think proper check here should be using this:
```
if not isinstance(item.get('releaseType', ""), str)
```
In other words `None` is not str data-type. For instance, releaseType may not be set in item, and it may be ok, but if you default to None this check will always fail since it `None` is not type of `str`

Reply:
This is exactly what I want to test here, when `releaseType` is not provided, an exception should be raised (because I want to make sure it's in the schema). In other words, if that key is not present, the if statement should evaluate to `True` and an exception raised.